json_rpc/clients/httpclient.nim

```diff
@@ -45,16 +45,34 @@ method call*(client: RpcHttpClient, name: string,
     req = HttpClientRequestRef.post(client.httpSession,
                                     client.httpAddress.get,
                                     body = reqBody.toOpenArrayByte(0, reqBody.len - 1))
-    res = await req.send()
+    res =
+      try:
+        await req.send()
+      except CancelledError as exc:
+        raise exc
+      except HttpError as exc:
+        raise exc
```

Comment:
What's the idea of this code? Catching and re-raising the exception is functionally equivalent to not catching them at all.

Reply:
47dcaad2d32cdedfd5ed5c3a83a5f67f58f51f22
this approach is also very simplistic

if still too naive, please give me some pointers on the right direction